### PR TITLE
Add: M3-2385 Ability to Remove Public IP Addresses

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Adds the ability to remove public IP Addresses

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers

Did some refactoring to flatten state inside the LinodeNetworking componens

### To test

Use prod-test-003 and use Linode "has_many_pub_ips" and try deleting some IPs

### CODE REVIEW

**Comment by martinmckenna** [comment]
do not merge until after Linodes detail refactor is merged

**Comment by martinmckenna** [comment]
note to self: this should most likely update the redux store once an IP is removed

File: src/features/linodes/LinodesDetail/LinodeNetworking/DeleteIPConfirm.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            ...linode,
+            ipv4: linode.ipv4.filter(eachIP => eachIP !== data.IPAddress)
+          };
+          ipRemoveSuccess(linodeWithRemovedIP);
+          handleClose();
+        }
+      })
+      .catch(e => {
+        const errText = pathOr(
+          'There was an error removing this IP. Please try again later',
```
Comment (Jskobos) [inline src/features/linodes/LinodesDetail/LinodeNetworking/DeleteIPConfirm.tsx]: later.

File: src/features/linodes/LinodesDetail/LinodeNetworking/DeleteIPConfirm.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          const linodeWithRemovedIP = {
+            ...linode,
+            ipv4: linode.ipv4.filter(eachIP => eachIP !== data.IPAddress)
+          };
+          ipRemoveSuccess(linodeWithRemovedIP);
+          handleClose();
+        }
+      })
+      .catch(e => {
+        const errText = pathOr(
```
[the remark this answers is not in the document]
Comment (Jskobos) [reply src/features/linodes/LinodesDetail/LinodeNetworking/DeleteIPConfirm.tsx]: (imo consumers at this level should only be getting API errors, never Axios errors)

File: src/services/linodes/linodeIPs.ts [hunk 3]
```diff
@@ -43,3 +43,26 @@ export const allocateIPAddress = (
...
+  IPAddress: string;
+}) => {
+  return Request<{}>(
+    setURL(
+      `${API_ROOT}/linode/instances/${payload.linodeID}/ips/${
+        payload.IPAddress
+      }`
+    ),
+    setMethod('DELETE')
+  ).then(response => response.data);
```
Comment (Jskobos) [inline src/services/linodes/linodeIPs.ts]: This doesn't really do anything, since the response is empty

**Pull request merged**